jobs/build.Jenkinsfile

```diff
@@ -552,3 +557,22 @@ def run_release_job(buildID) {
         ]
     }
 }
+
+// XXX: generalize and put in coreos-ci-lib eventually
+def wait_until_locked_or_continue(resource, timeout_mins) {
+    try {
+        timeout(time: timeout_mins, unit: 'MINUTES') {
+            waitUntil {
+                lock(resource: resource, skipIfLocked: true) {
+                    return false
+                }
+                return true
+            }
+        }
+    } catch (FlowInterruptedException e) {
+        // If the lock was still not taken, then something went wrong. We
+        // continue to try to at least release for the arches that did succeed.
+        // We may be able to salvage the failed arch in the next run.
```

Comment:
This comment is leaking context in here about why it's OK for the caller to continue, which shouldn't matter inside this function (i.e. what if this function was called by more than one caller.

i.e. this comment should be something more like:

```
// The lock wasn't taken in the allotted time, but the caller
// wants us to continue anyway.

Reply:
This is why there's a comment above the helper calling to generalize it and moving it to coreos-ci-lib. :) Just didn't want to block on that since I think we should gain experience with it first and see if there are any issues.